### PR TITLE
Port Verifier, Session and ProjectDeployer

### PR DESCRIPTION
**Pull request by jcarpanelli**
Related to #483 
Ports:
- [x] Verifier
- [x] ProjectDeployer
- [x] Session

### CODE REVIEW

File: packages/cli/src/models/network/ProjectDeployer.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  constructor(controller: NetworkController, requestedVersion: string) {
+    this.controller = controller;
+    this.packageFile = controller.packageFile;
+    this.networkFile = controller.networkFile;
+    this.txParams = controller.txParams;
+    this.requestedVersion = requestedVersion;
+  }
+}
+
+class BasePackageProjectDeployer extends BaseProjectDeployer {
```
Comment (ajsantander) [inline packages/cli/src/models/network/ProjectDeployer.ts]: :heart_eyes:

File: packages/cli/src/models/network/ProjectDeployer.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  get packageAddress(): string {
+    return this.controller.packageAddress;
+  }
+
+  protected _tryRegisterPartialDeploy({ thepackage, directory }: PartialDeploy): void {
+    if (thepackage) this._registerPackage(thepackage);
+    if (directory) this._registerVersion(this.requestedVersion, directory);
+  }
+
+  protected _registerPackage({ address }: { address: string }): void {
```
Comment (ajsantander) [inline packages/cli/src/models/network/ProjectDeployer.ts]: Probably not worth doing anything atm, but note that `{ address }` is the de-structuring of an incoming object that has a property "address" (and could have any other properties). OTOH, the typing `{ address: string }` is a type that defines an object that _only_ has the property "address". 

I'm not a 100% sure about this, but let's just keep this in the back of our heads for now.

File: packages/cli/src/models/network/ProjectDeployer.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    return this._run((existingAddresses: ExistingAddresses) => (
+      AppProject.fromSimpleProject(simpleProject, this.requestedVersion, existingAddresses)
+    ));
+  }
+
+  get appAddress(): string {
+    return this.controller.appAddress;
+  }
+
+  public async _run(createProjectFn: CreateProjectFn): Promise<AppProject | never> {
```
Comment (ajsantander) [inline packages/cli/src/models/network/ProjectDeployer.ts]: private?

Comment (jcarpanelli) [reply packages/cli/src/models/network/ProjectDeployer.ts]: nice catch :slightly_smiling_face:

File: packages/cli/src/models/network/ProjectDeployer.ts (hunk 2 above)
Comment (jcarpanelli) [reply packages/cli/src/models/network/ProjectDeployer.ts]: Already discussed it offline.

**Pull request merged**